### PR TITLE
[Firewall] Add dns proxy settings for network firewall

### PR DESCRIPTION
**Pull request by haroldrandom**
Close https://github.com/Azure/azure-cli/issues/13350

add `--dns-servers`, `--dns-proxy-enabled`, `--dns-require-proxy-for-network-rules` to `network firewall create/update`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S171

File: src/azure-firewall/azext_firewall/custom.py [hunk 1]
```diff
@@ -80,11 +81,17 @@ def create_azure_firewall(cmd, resource_group_name, azure_firewall_name, locatio
         if firewall.additional_properties is None:
             firewall.additional_properties = {}
         firewall.additional_properties['Network.SNAT.PrivateRanges'] = private_ranges
+
+    firewall.additional_properties['DNSEnableProxy'] = dns_proxy_enabled or False
+    firewall.additional_properties['DNSRequireProxyForNetworkRules'] = dns_require_proxy_for_network_rules or True
```
Comment (MyronFanQiu) [inline src/azure-firewall/azext_firewall/custom.py:86]: It's always True

Comment (haroldrandom) [reply src/azure-firewall/azext_firewall/custom.py]: fixed

File: src/azure-firewall/azext_firewall/_params.py [hunk 2]
```diff
@@ -56,6 +57,12 @@ def load_arguments(self, _):
                                                                                   'The default sku in server end is AZFW_VNet. '
                                                                                   'If you want to attach azure firewall to vhub, you should set sku to AZFW_Hub.')
         c.argument('private_ranges', nargs='+', validator=process_private_ranges, help='Space-separated list of SNAT private range. Validate values are single Ip, Ip prefixes or a single special value "IANAPrivateRanges"')
+
+    with self.argument_context('network firewall', arg_group='DNS') as c:
+        c.argument('dns_servers', nargs='+', help='Space-separated list of DNS server IP addresses')
+        c.argument('dns_proxy_enabled', arg_type=get_three_state_flag(), help='Enable DNS Proxy')
```
Comment (MyronFanQiu) [inline src/azure-firewall/azext_firewall/_params.py:63]: better to be `enable_dns_proxy`

Comment (haroldrandom) [reply src/azure-firewall/azext_firewall/_params.py:63]: changed.

**Comment by vbrinza** [comment]
in which version of the extension is the option available ?

**Comment by haroldrandom** [comment]
> in which version of the extension is the option available ?

Hi, @vbrinza , we are experiencing an regression after bringing this feature. We are proactively connecting with Service Team to fix it.

Will let you know once we reach an conclusion ASAP.

**Comment by vbrinza** [comment]
@haroldrandom is there any progress on this? Thanks.

**Comment by jsntcy** [comment]
@msyyc, please help check if this feature is released or not.

**Comment by msyyc** [comment]
Ok, I will check it soon.

**Comment by msyyc** [comment]
@vbrinza Please use 'az extension add --name azure-firewall ' to try the latest version

**Pull request merged**